eth2/utils/bls/src/aggregate_signature.rs

```diff
@@ -1,30 +1,44 @@
-use super::{AggregatePublicKey, Signature};
+use super::{AggregatePublicKey, Signature, BLS_AGG_SIG_BYTE_SIZE};
 use bls_aggregates::{
     AggregatePublicKey as RawAggregatePublicKey, AggregateSignature as RawAggregateSignature,
 };
 use serde::de::{Deserialize, Deserializer};
 use serde::ser::{Serialize, Serializer};
-use serde_hex::{encode as hex_encode, PrefixedHexVisitor};
-use ssz::{
-    decode_ssz_list, hash, ssz_encode, Decodable, DecodeError, Encodable, SszStream, TreeHash,
-};
+use serde_hex::{encode as hex_encode, HexVisitor};
+use ssz::{decode, hash, Decodable, DecodeError, Encodable, SszStream, TreeHash};
 
 /// A BLS aggregate signature.
 ///
 /// This struct is a wrapper upon a base type and provides helper functions (e.g., SSZ
 /// serialization).
 #[derive(Debug, PartialEq, Clone, Default, Eq)]
-pub struct AggregateSignature(RawAggregateSignature);
+pub struct AggregateSignature {
+    aggregate_signature: RawAggregateSignature,
+    is_empty: bool,
+}
 
 impl AggregateSignature {
     /// Instantiate a new AggregateSignature.
+    ///
+    /// is_empty is false
+    /// AggregateSiganture is point at infinity
     pub fn new() -> Self {
-        AggregateSignature(RawAggregateSignature::new())
+        Self {
+            aggregate_signature: RawAggregateSignature::new(),
+            is_empty: false,
+        }
     }
 
     /// Add (aggregate) a signature to the `AggregateSignature`.
     pub fn add(&mut self, signature: &Signature) {
-        self.0.add(signature.as_raw())
+        if !self.is_empty {
+            self.aggregate_signature.add(signature.as_raw())
+        }
+    }
+
+    /// Add (aggregate) another `AggregateSignature`.
+    pub fn add_aggregate(&mut self, agg_signature: &AggregateSignature) {
+        self.0.add_aggregate(&agg_signature.0)
```

Comment:
This won't compile because the field was renamed (a semantic merge conflict). I've fixed it locally, do you want me to push?

Reply:
Fixed it thanks :)